.github/workflows/deploy.yaml

```
@@ -0,0 +1,51 @@
+name: Build and Deploy to Netlify
+
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
```

The GitHub Actions docs for Python also recommend caching dependencies to speed up builds:

```suggestion
          python-version: '3.13'
          cache: 'pip'
```

Reference: https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-python#caching-dependencies